models/keyword_test.go

```diff
@@ -69,65 +63,112 @@ func (s *KeywordDBTestSuite) TestSaveKeywordsWithValidParams() {
+	bulkData := []Keyword{
+		{Keyword: "", UserID: s.userID},
+	}
 
-	result, err := SaveKeywords(s.userID, keywords)
+	result, err := SaveKeywords(bulkData)
 
 	assert.Equal(s.T(), nil, err)
 	assert.Equal(s.T(), 1, len(result))
 	assert.Equal(s.T(), "", result[0].Keyword)
 }
 
 func (s *KeywordDBTestSuite) TestSaveKeywordsWithEmptySlice() {
-	keywords := []string{}
+	bulkData := []Keyword{}
 
-	result, err := SaveKeywords(s.userID, keywords)
+	result, err := SaveKeywords(bulkData)
+	_, isPgError := err.(*pgconn.PgError)
 
-	assert.Equal(s.T(), "invalid data", err.Error())
+	assert.Equal(s.T(), "empty slice found", err.Error())
+	assert.Equal(s.T(), false, isPgError)
 	assert.Equal(s.T(), nil, result)
 }
 
 func (s *KeywordDBTestSuite) TestSaveKeywordsWithInvalidUserID() {
-	keywords := []string{"Hazard"}
+	bulkData := []Keyword{
+		{Keyword: "Hazard", UserID: 99999999},
+	}
 
-	result, err := SaveKeywords(9999999999, keywords)
+	result, err := SaveKeywords(bulkData)
+	errVal, isPgError := err.(*pgconn.PgError)
 
-	assert.Equal(s.T(), "something went wrong, please try again", err.Error())
+	assert.Equal(s.T(), "23503", errVal.Code)
+	assert.Equal(s.T(), true, isPgError)
 	assert.Equal(s.T(), nil, result)
 }
 
-func TestValidateCSVLengthWithMinRowAllowed(t *testing.T) {
-	result := ValidateCSVLength(1)
+func (s *KeywordDBTestSuite) TestGetKeywordsByWithMoreThanOneRows() {
+	bulkData := []Keyword{
+		{Keyword: "Hazard", UserID: s.userID},
+		{Keyword: "Ronaldo", UserID: s.userID},
+		{Keyword: "Neymar", UserID: s.userID},
+		{Keyword: "Messi", UserID: s.userID},
+		{Keyword: "Mbappe", UserID: s.userID},
+	}
 
-	assert.Equal(t, nil, result)
-}
+	db.GetDB().Create(&bulkData)
 
-func TestValidateCSVLengthWithMaxRowAllowed(t *testing.T) {
-	result := ValidateCSVLength(1000)
+	result, err := GetKeywordsBy(nil)
 
-	assert.Equal(t, nil, result)
+	assert.Equal(s.T(), 5, len(result))
+	assert.Equal(s.T(), "Hazard", result[0].Keyword)
+	assert.Equal(s.T(), "Mbappe", result[1].Keyword)
+	assert.Equal(s.T(), "Messi", result[2].Keyword)
+	assert.Equal(s.T(), "Neymar", result[3].Keyword)
+	assert.Equal(s.T(), "Ronaldo", result[4].Keyword)
+	assert.Equal(s.T(), nil, err)
 }
 
-func TestValidateCSVLengthWithZeroRow(t *testing.T) {
-	result := ValidateCSVLength(0)
+func (s *KeywordDBTestSuite) TestGetKeywordsByWithTruthyCondition() {
+	keyword := Keyword{UserID: s.userID, Keyword: faker.Name()}
+	db.GetDB().Create(&keyword)
 
-	assert.Equal(t, "CSV file must contain between 1 to 1000 keywords", result.Error())
+	condition := make(map[string]interface{})
+	condition["keyword"] = keyword.Keyword
+
+	result, err := GetKeywordsBy(condition)
+
+	assert.Equal(s.T(), 1, len(result))
+	assert.Equal(s.T(), keyword.Keyword, result[0].Keyword)
+	assert.Equal(s.T(), nil, err)
 }
 
-func TestValidateCSVLengthWithGreaterThanMaxRowAllowed(t *testing.T) {
-	result := ValidateCSVLength(1001)
+func (s *KeywordDBTestSuite) TestGetKeywordsByWithFalsyCondition() {
```

Comment:
I am not sure that the new method name `FalsyCondition`, `NilCondition`... help. I have to go check each test very much in details to understand the naming 🤔

Reply:
Could you please explain more about this?

Reply:
When reading the test method name `TestGetKeywordsByWithFalsyCondition`, I don't understand which case we are testing. So the name is not descriptive enough. That test is for retrieving the list of keywords when a keyword does not exist. So `TestGetKeywordsByWithInvalidKeyword` would be easier to read and understand. Same for `TestGetKeywordsByWithTruthyCondition` which could be replaced with `TestGetKeywordsByWithValidKeyword`. And `TestGetKeywordsByWithNilCondition` could be replaced with `TestGetKeywordsByWithBlankKeyword` or `TestGetKeywordsByWithoutKeyword`.

Reply:
Thanks for clarifying. Fixed in 656fd51a651600d836bf427be928ffe27c4f83ad